### PR TITLE
Add initial implementation for automoderator api

### LINKED ISSUE #441: Support the automoderator API
Discord has an api to interact with automoderator. It would be nice if Arikawa would implement this api.

https://discord.com/developers/docs/resources/auto-moderation#list-auto-moderation-rules-for-guild

### PR DESCRIPTION
**Pull request by hi117**
Closes #441

### CODE REVIEW

File: discord/automoderation.go [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+type AutoModerationKeywordPresetTypes uint32
+
+const (
+	AutoModeratorProfanity = 1 + iota
+	AutoModeratorSexualContent
+	AutoModeratorSlurs
+)
+
+type AutoModerationRuleID Snowflake
```
Comment (diamondburned) [inline discord/automoderation.go]: For the snowflake types, you'd want to add it into `snowflake.go`'s `go:generate` line the re-generate the code via `go generate ./...`.

File: api/automoderation.go [hunk 2]
```diff
@@ -0,0 +1,51 @@
+package api
+
+import (
+	"github.com/diamondburned/arikawa/v3/discord"
+	"github.com/diamondburned/arikawa/v3/utils/httputil"
+)
+
+func (c *Client) ListAutoModerationRules(guildID discord.GuildID) ([]discord.AutoModerationRule, error) {
```
Comment (diamondburned) [inline api/automoderation.go:12]: Could you put copy-pasted Discord API documentation comments here for convenience?

**Review by diamondburned — APPROVED** [review]

**Review by diamondburned — CHANGES_REQUESTED** [review]
sorry, eepy rn...

File: discord/automoderation.go (hunk 1 above)
Comment (hi117) [reply discord/automoderation.go]: Done in [e1b274b](https://github.com/diamondburned/arikawa/pull/444/commits/e1b274bc74e9af7d3ee1fa6bef8cfc2a73f644ce)

File: api/automoderation.go (hunk 2 above)
Comment (hi117) [reply api/automoderation.go:12]: Done in [35c61e8](https://github.com/diamondburned/arikawa/pull/444/commits/35c61e83a30d37a5d42e43af925be5537893ff04)

File: api/automoderation.go [hunk 3]
```diff
@@ -0,0 +1,72 @@
+package api
+
+import (
+	"github.com/diamondburned/arikawa/v3/discord"
+	"github.com/diamondburned/arikawa/v3/utils/httputil"
+)
+
+// Get a list of all rules currently configured for the guild. Returns a list of auto moderation rule objects for the given guild.
```
Comment (diamondburned) [inline api/automoderation.go]: Please edit these comments to be in the format `// ListAutoModerationRules gets a list of`...

File: api/automoderation.go [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+}
+
+// Modify an existing rule. Returns an auto moderation rule on success. Fires an Auto Moderation Rule Update Gateway event.
+//
+// Requires MANAGE_GUILD permissions.
+//
+// All parameters for this endpoint are optional.
+//
+// This endpoint supports the X-Audit-Log-Reason header.
+func (c *Client) ModifyAutoModerationRule(guildID discord.GuildID, ruleID discord.AutoModerationRuleID, rule discord.AutoModerationRule, data ModifyAutoModerationRuleData) (*discord.AutoModerationRule, error) {
```
Comment (diamondburned) [inline api/automoderation.go]: For this, you should do::

```go
type ModifyAutoModerationRuleData struct {
	discord.AutoModerationRule
	AuditLogReason
}

func (c *Client) ModifyAutoModerationRule(data ModifyAutoModerationRuleData) (*discord.AutoModerationRule, error)
```

and take all the other parameters from `data`.

File: api/automoderation.go [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+type DeleteAutoModerationRuleData struct {
+	AuditLogReason
+}
+
+// Delete a rule. Returns a 204 on success. Fires an Auto Moderation Rule Delete Gateway event.
+//
+// This endpoint requires the MANAGE_GUILD permission.
+//
+// This endpoint supports the X-Audit-Log-Reason header.
+func (c *Client) DeleteAutoModerationRule(guildID discord.GuildID, ruleID discord.AutoModerationRuleID, data DeleteAutoModerationRuleData) error {
```
Comment (diamondburned) [inline api/automoderation.go]: You can write

```go
type DeleteAutoModerationRuleData struct {
	GuildID discord.GuildID
	RuleID  discord.AutoModerationRuleID
	AuditLogReason
}

func (c *Client) DeleteAutoModerationRule(data DeleteAutoModerationRuleData) error
```

for this as well.

**Review by diamondburned — APPROVED** [review]

**Review by diamondburned — CHANGES_REQUESTED** [review]
aghh i did it again

File: api/automoderation.go (hunk 3 above)
Comment (hi117) [reply api/automoderation.go]: Updated in [5ac59ad](https://github.com/diamondburned/arikawa/pull/444/commits/5ac59ad300e1c113bfa99d7cf868b05079647505)

File: api/automoderation.go (hunk 4 above)
Comment (hi117) [reply api/automoderation.go]: Updated in [5ac59ad](https://github.com/diamondburned/arikawa/pull/444/commits/5ac59ad300e1c113bfa99d7cf868b05079647505)

File: api/automoderation.go (hunk 5 above)
Comment (hi117) [reply api/automoderation.go]: Updated in [5ac59ad](https://github.com/diamondburned/arikawa/pull/444/commits/5ac59ad300e1c113bfa99d7cf868b05079647505)

**Review by diamondburned — APPROVED** [review]

**Pull request merged**